ArduPlane/takeoff.cpp

```diff
@@ -271,13 +271,9 @@ int8_t Plane::takeoff_tail_hold(void)
  */
 void Plane::complete_auto_takeoff(void)
 {
-#if GEOFENCE_ENABLED == ENABLED
-    if (g.fence_autoenable > 0) {
-        if (! geofence_set_enabled(true)) {
-            gcs().send_text(MAV_SEVERITY_NOTICE, "Enable fence failed (cannot autoenable");
-        } else {
-            gcs().send_text(MAV_SEVERITY_INFO, "Fence enabled (autoenabled)");
-        }
+#if AC_FENCE == ENABLED
+    if(fence.auto_enabled() >= AC_Fence::AutoEnable::ALWAYS_ENABLED) {
```

Comment:
No~

```suggestion
    if(fence.auto_enabled() == AC_Fence::AutoEnable::ALWAYS_ENABLED) {
```

... and add in any other values you need....

Reply:
Updated, and likened with landing implementation - that is `switch` on specific auto conditions.